server/container/guice/guice-common/src/main/java/org/apache/james/modules/protocols/SMTPServerModule.java

```diff
@@ -42,6 +45,9 @@
 
     @Override
     protected void configure() {
+        bind(DNSServiceXBillImpl.class).in(Scopes.SINGLETON);
```

Comment:
This DNSService is not the one introduced in the DNSModule we have somewhere else.
This one is a specific, JSPF concern. I would not put it there, because having two very different DNSService in the same module seem confusing.